pyo3-arrow/src/table.rs

```diff
@@ -399,6 +399,33 @@ impl PyTable {
         Ok(Self::try_new(batches, schema)?)
     }
 
+    fn drop_columns(&self, fields: Vec<String>) -> PyArrowResult<Arro3Table> {
```

Comment:
This parameter is the same parameter passed to pyo3 as the keyword name. So if you passed in `Table.drop_columns(columns=...)` it would fail and expect `fields=` instead. 

We should probably rename it to `names` instead to be consistent with pyarrow: https://arrow.apache.org/docs/python/generated/pyarrow.Table.html#pyarrow.Table.rename_columns (and we need to be consistent with the name between the Rust implementation and the pyi file)

```suggestion
    fn drop_columns(&self, names: Vec<String>) -> PyArrowResult<Arro3Table> {
```

Also as a side note I see that pyarrow also allows `dict[str, str]`, which we could support as well (in a future PR is fine)